hudi-client/src/test/java/org/apache/hudi/client/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -399,56 +405,173 @@ public void testDeletes() throws Exception {
   }
 
   /**
-   * Test update of a record to different partition with Global Index.
+   * Tests when update partition path is set in global bloom, existing record in old partition
+   * is deleted appropriately.
+   * @throws Exception
    */
   @Test
-  public void testUpsertToDiffPartitionGlobalIndex() throws Exception {
-    HoodieWriteClient client = getHoodieWriteClient(getConfig(IndexType.GLOBAL_BLOOM), false);
-    /**
-     * Write 1 (inserts and deletes) Write actual 200 insert records and ignore 100 delete records
-     */
-    String newCommitTime = "001";
-    List<HoodieRecord> inserts1 = dataGen.generateInserts(newCommitTime, 100);
+  public void testUpsertsUpdatePartitionPathRegularGlobalBloom() throws Exception {
+    testUpsertsUpdatePartitionPathGlobalBloom(IndexType.GLOBAL_BLOOM, getConfig(),
+        HoodieWriteClient::upsert);
+  }
+
+  /**
+   * Tests when update partition path is set in simple global bloom, existing record in
+   * old partition is deleted appropriately.
+   * @throws Exception
+   */
+  @Test
+  public void testUpsertsUpdatePartitionPathSimpleGlobalBloom() throws Exception {
+    testUpsertsUpdatePartitionPathGlobalBloom(IndexType.GLOBAL_SIMPLE, getConfig(),
+        HoodieWriteClient::upsert);
+  }
+
+  private void testUpsertsUpdatePartitionPathGlobalBloom(IndexType indexType,
+      HoodieWriteConfig config,
+      Function3<JavaRDD<WriteStatus>, HoodieWriteClient, JavaRDD<HoodieRecord>, String> writeFn)
+      throws Exception {
+    // Force using older timeline layout
```

Comment:
why is this relevant for this test?

Reply:
I don't know actually. I tried removing version_0 and the test fails. If we know why testUpsertsInternal in the same class does it (@bvaradar ), we might have an answer.